sympy/printing/str.py

```diff
@@ -74,13 +74,20 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+    def _print_Not(self, expr):
+        return '~%s' %(self.parenthesize(expr.args[0],PRECEDENCE["Not"]))
+
     def _print_And(self, expr):
-        return '%s(%s)' % (expr.func, ', '.join(sorted(self._print(a) for a in
-            expr.args)))
+        return self.stringify(expr.args, " & ", 38)
+    # The number 38 has been used (instead of PRECEDENCE["And"]) as
```

Comment:
It would be better to add something to the precedence dict, rather than using a magic number here.